modules/RTC/TPCUtils.js

```diff
@@ -191,12 +191,12 @@ export class TPCUtils {
     /**
     * Adds {@link JitsiLocalTrack} to the WebRTC peerconnection for the first time.
     * @param {JitsiLocalTrack} track - track to be added to the peerconnection.
-    * @param {boolean} isInitiator - boolean that indicates if the endpoint is offerer
-    * in a p2p connection.
-    * @returns {void}
+    * @param {boolean} isInitiator - boolean that indicates if the endpoint is offerer in a p2p connection.
+    * @returns {Promise<void>} - resolved when done, rejected otherwise.
```

Comment:
This code is not really async. Note that you're returning either resolve or reject immediately. Is that necessary?

Reply:
Yeah, I didn't have to return a promise here. I am catching the error in TPC#addTrack now.